addons/mail/models/mail_activity.py

```diff
@@ -571,53 +584,160 @@ def action_open_document(self):
     def activity_format(self):
         activities = self.read()
         self.mail_template_ids.fetch(['name'])
+        self.attachment_ids.fetch(['name'])
         for record, activity in zip(self, activities):
             activity['mail_template_ids'] = [
                 {'id': mail_template.id, 'name': mail_template.name}
                 for mail_template in record.mail_template_ids
             ]
+            activity['attachment_ids'] = [
+                {'id': attachment.id, 'name': attachment.name}
+                for attachment in record.attachment_ids
+            ]
         return activities
 
     @api.model
-    def get_activity_data(self, res_model, domain, limit=None, offset=0):
+    def get_activity_data(self, res_model, domain, limit=None, offset=0, fetch_done=False):
+        """ Get aggregate data about records and their activities.
+
+        The goal is to display aggregated data about records and their activities for
+        example in activity views and the chatter.
+
+        The aggregated data returned are based on the records along with their related
+        activities that are fetched according to the search criterion: res_model,
+        domain, limit, offset and fetch_done. The fetch_done parameter determines if
+        "done" activity are integrated in the aggregated data or not.
+
+        :param str res_model: model of the records to fetch
+        :param list domain: record search domain
+        :param int limit: maximum number of records to fetch
+        :param int offset: offset of the first record to fetch
+        :return dict: {'activity_types': list of activity type info as list
```

Comment:
`fetch_done` is missing before that line
(you can move the explanation above here)